qcodes/instrument_drivers/tektronix/keithley_7510.py

```diff
@@ -1,10 +1,367 @@
+    value = None
+
+    def __init__(self,
+                 names: tuple,
+                 shapes: tuple,
+                 setpoints: tuple,
+                 **kwargs):
+        self.names = names
+        super().__init__(name='data_array_7510',
+                         names=names,
+                         shapes=shapes,
+                         setpoints=setpoints,
+                         **kwargs)
+        for param_name in self.names:
+            self.__dict__.update({param_name: []})
+
+    def set_raw(self, value: tuple) -> None:
+        self.value = value
+        for i in range(len(self.names)):
+            setattr(self, self.names[i], value[i])
+
+    def get_raw(self) -> Optional[tuple]:
+        return self.value
+
+
+class GeneratedSetPoints(Parameter):
+    """
+    A parameter that generates a setpoint array from start, stop and num points
+    parameters.
+    """
+    def __init__(self,
+                 start: Parameter,
+                 stop: Parameter,
+                 n_points: Parameter,
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._start = start
+        self._stop = stop
+        self._n_points = n_points
+
+    def get_raw(self):
+        return np.linspace(self._start(), self._stop(), self._n_points())
+
+
+class ParameterWithSetpointsCustomized(ParameterWithSetpoints):
+    """
+    A Parameter class which can includes string values.
+    """
+    def validate(self, value: Any) -> None:
+        """
+        Overwrites the standard ``validate`` method to only check the the
+        parameter has consistent shape with its setpoints.
+
+        Note: the validation of the datatype is removed!
+        """
+        self.validate_consistent_shape()
+
+
+class Buffer7510(InstrumentChannel):
+    """
+    Treat the reading buffer as a submodule, similar to Sense.
+    """
+    default_buffer = {"defbuffer1", "defbuffer2"}
+
+    buffer_elements = {
+        "date": "DATE",
+        "measurement_formatted": "FORMatted",
+        "fractional_seconds": "FRACtional",
+        "measurement": "READing",
+        "relative_time": "RELative",
+        "seconds": "SEConds",
+        "source_value": "SOURce",
+        "source_value_formatted": "SOURFORMatted",
+        "source_value_status": "SOURSTATus",
+        "source_value_unit": "SOURUNIT",
+        "measurement_status": "STATus",
+        "time": "TIME",
+        "timestamp": "TSTamp",
+        "measurement_unit": "UNIT"
+    }
+
+    inverted_buffer_elements = invert_val_mapping(buffer_elements)
+
+    def __init__(
+            self,
+            parent: 'Keithley7510',
+            name: str,
+            size: Optional[int] = None,
+            style: str = ''
+    ) -> None:
+        super().__init__(parent, name)
+        self._size = size
+        self.style = style
+        self.data_start = 1  # first index of the data to be returned
+        self.data_end = 1    # last index of the data to be returned
```

Comment:
Please make data_end and data_start parameters with get_cmd and set_cmd = False.

Reply:
They are parameters now, with get/set_cmd=`None` (I have been using `None` for other places, but it has to be `False`, please let me know)